apps/trading/components/market-banner/market-termination-banner.spec.tsx

```diff
@@ -0,0 +1,129 @@
+
+const walletMock = {
+  pubKey: 'pubKey',
+  pubKeys: [{ publicKey: 'pubKey' }, { publicKey: 'secondPubKey' }],
+};
+jest.mock('@vegaprotocol/wallet', () => ({
+  ...jest.requireActual('@vegaprotocol/wallet'),
+  useVegaWallet: jest.fn(() => walletMock),
+}));
+
+const proposalMock: MockedResponse<TerminateProposalsListQuery> = {
+  request: {
+    query: TerminateProposalsListDocument,
+    variables: undefined,
+  },
+  result: {
+    data: {
+      proposalsConnection: {
+        edges: [
+          {
+            node: {
+              id: 'first-id',
+              terms: {
+                closingDatetime: '2023-09-27T11:48:18Z',
+                enactmentDatetime: '2023-09-30T11:48:18',
+                change: {
+                  __typename: 'UpdateMarketState',
+                  updateType: '',
+                  price: '',
+                  market: {
+                    id: 'market-1',
+                    tradableInstrument: {
+                      instrument: {
+                        name: 'Market one',
+                      },
+                    },
+                  },
+                },
+              },
+            },
+          },
+          {
+            node: {
+              id: 'second-id',
+              terms: {
+                closingDatetime: '2023-09-27T11:48:18Z',
+                enactmentDatetime: '2023-10-01T11:48:18',
+                change: {
+                  __typename: 'UpdateMarketState',
+                  updateType: '',
+                  price: '',
+                  market: {
+                    id: 'market-2',
+                    tradableInstrument: {
+                      instrument: {
+                        name: 'Market two',
+                      },
+                    },
+                  },
+                },
+              },
+            },
+          },
+        ],
+      },
+    } as unknown as TerminateProposalsListQuery,
+  },
+};
+const positionsMock: MockedResponse<PositionsQuery> = {
+  request: {
+    query: PositionsDocument,
+    variables: { partyIds: ['pubKey', 'secondPubKey'] },
+  },
+  result: {
+    data: {
+      positions: {
+        edges: [
+          {
+            node: {
+              market: {
+                id: 'market-1',
+              },
+              party: {
+                id: 'pubKey',
+              },
+            },
+          },
+          {
+            node: {
+              market: {
+                id: 'market-2',
+              },
+              party: {
+                id: 'secondPubKey',
+              },
+            },
+          },
+        ],
+      },
+    } as unknown as PositionsQuery,
+  },
+};
+const mocks: MockedResponse[] = [proposalMock, positionsMock];
+
+describe('MarketTerminationBanner', () => {
+  it('should be properly rendered', async () => {
+    render(
+      <MockedProvider mocks={mocks}>
+        <MarketTerminationBanner />
+      </MockedProvider>
+    );
+    await waitFor(() => {
+      expect(
```

Comment:
The following is more succinct

expect(await screen.findByTestId('...')).toBeInTheDocument()

Reply:
`await` in `expect` is not enough to let `MockedProvider` for resolving its mocks, probably since we have there two requests one after another